examples/spellchecker/src/main.rs

```diff
@@ -42,8 +38,7 @@ fn main() -> windows::Result<()> {
         if result == S_FALSE {
             break;
         }
-        result.ok()?;
-        let error = error.unwrap();
+        let error = result.and_some(error)?;
```

Comment:
To me, this change made the code a bit harder to comprehend, though I still wanted to get rid of the `unwrap()`. Any suggestions on getting the readability back and dropping the `unwrap()` at the same time?

Reply:
I wouldn't worry too much about this. I plan to make these work more like WinRT methods where a `Result<T>` is returned and trailing `retval` parameters are turned into return types. #509 - that should simplify this a bunch and make the whole `and_some` pattern less essential.

Reply:
Just as a thought - would renaming `and_some` to something else potentially make this easier? The closest equivalent in `std` IMO is `Option::and`. Is `result.and(error)?` any clearer?

Reply:
Thanks for the feedback, Kenny. With that I believe you can merge this PR as you see fit.

Looking forward to a more compact and ergonomic interface over COM's `retval`'s, although I'm not sure how this is going to pan out. As far as I understand, the Windows Runtime uses a single success value (`S_OK`), making this easy. With COM there are more success codes in use (such as `S_FALSE` as in this case, the reason why this part of the code needs to manually check the return value in the first place).

Reply:
Dealing with other success codes will be an issue, but as long as the original HRESULT can still be retrieved I think we should be fine.